ArmPkg/ArmPkg.dec

```diff
@@ -328,6 +328,7 @@
   gArmTokenSpaceGuid.PcdGenericWatchdogControlBase|0x2A440000|UINT64|0x00000007
   gArmTokenSpaceGuid.PcdGenericWatchdogRefreshBase|0x2A450000|UINT64|0x00000008
   gArmTokenSpaceGuid.PcdGenericWatchdogEl2IntrNum|93|UINT32|0x00000009
+  gArmTokenSpaceGuid.PcdGenericWatchdogInstallInterruptHandler|TRUE|BOOLEAN|0x0000000A

```

Comment:
Sorry to bikeshed but could we choose a better name for this? It is not obvious from the name what this does, which is to reset the system when the watchdog timer expires. So PcdGenericWatchdogResetSystemOnTimeoutExpire might be a better name (but feel free to be creative and come up with something better)

Also, could you explain why a platform would elect to use this driver if its SCP is in charge of this? Isn't that a layering violation, and wouldn't it make more sense for the SCP to notify the CPU if such a notification is even needed?

Reply:
> Also, could you explain why a platform would elect to use this driver if its SCP is in charge of this? Isn't that a layering violation, and wouldn't it make more sense for the SCP to notify the CPU if such a notification is even needed?

@ardbiesheuvel I'm unsure I fully understand your suggestion. It's necessary to configure the watchdog timer period through this DXE driver to monitor the UEFI boot process. The SCP is only involved with the watchdog when the timeout happens to facilitate logging. The action could be IMPLEMENTATION DEFINED. Given the watchdog timer protocol is architecturally required in UEFI, the proposed PCD here is to offer platforms the flexibility to implement their preferred action for handling the watchdog interrupt.

According to [Arm Base System Architecture](https://developer.arm.com/documentation/den0094/latest/), the generic Watchdog Timer provides two interrupts:
  * Watchdog Signal 0 (WS0) – this is sometimes referred to as a “bark” interrupt. It is typically routed as an SPI to the GIC. If the Generic Watchdog is not refreshed periodically, it will raise a signal (WS0).
  * Watchdog Signal 1 (WS1) – this is sometimes referred to as a “bite” interrupt. It can be used to interrupt higher-privileged software or cause a PE reset. If this watchdog remains un-refreshed (unresponsive), it will raise a second signal (WS1).

The GenericWatchdogDxe handles the WS0 and performs a system reset upon timeout. However, I wonder whether we could simply ignore the WS0 and wait for the WS1 to do IMP_DEF action?